liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/UndeployAppMojo.java

```diff
@@ -30,42 +32,37 @@
 /**
  * Undeploy application from liberty server. If no parameters have been defined
  * the mojo will undeploy all applications from the server.
- *
- * @goal undeploy
  */
+@Mojo(name = "undeploy")
+
 public class UndeployAppMojo extends BasicSupport {
     
     /**
      * A file name which points to a specific module's jar | war | ear | eba |
      * zip archive.
-     *
-     * @parameter expression="${appArchive}"
-     * @optional
      */
+    @Parameter(property = "appArchive")
     protected String appArchive = null;
     
     /**
      * Maven coordinates of an application to undeploy. This is best listed as a
      * dependency, in which case the version can be omitted.
-     *
-     * @parameter
      */
+    @Parameter
     protected ArtifactItem appArtifact;
     
     /**
      * A set of patterns to include or exclude from the undeploy operation. If
      * appArchive or appArtifact has been defined then this parameter will be
      * ignored.
-     *
-     * @parameter
      */
+    @Parameter(property = "patternSet")
```

Comment:
I don't see there is an @expresssion annotation in the original javadoc, are you trying to fix another issue here?